### PR TITLE
Add update docs script to CI

### LINKED ISSUE #8835: Auto-Generate JSON-RPC API docs
Originally from https://github.com/paritytech/wiki/issues/58 but transferred here for better visibility.

We have 2 tools
1. Generate markdown docs from js files (should be run during the CI to make sure we document methods in JS format)
1.  Make sure that no rust RPC is missing in JS files (can be automated to make a PR to wiki for example.)

@tomusdrw  can you add anything relevant for this issue to be picked up as a task or bounty?

### PR DESCRIPTION
**Pull request by cpurta**
Added a script to CI that will use the jsonrpc tool to update rpc
documentation then commit and push those to the wiki repo.

Resolves #8835

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -220,6 +221,16 @@ test-rust-nightly:
     - rust
     - rust-nightly
   allow_failure: true
+json-rpc-docs:
+  stage: docs
+  only:
+    - tags
+    - master
```
Comment (5chdn) [inline .gitlab-ci.yml]: might be sufficient to have this on `tags` only.

File: scripts/gitlab-rpc-docs.sh [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+update_wiki_docs() {
+    for file in $(ls jsonrpc/docs); do
+        module_name=${file:0:-3}
+        mv jsonrpc/docs/$file wiki/JSONRPC-$module_name-module.md
+    done
+}
+
+setup_git() {
+    git config user.email "runner@parity.com"
+    git config user.name "Parity Runner"
```
Comment (5chdn) [inline scripts/gitlab-rpc-docs.sh]: Does this user exist? Do we need to create some user that has access?

Comment (cpurta) [reply scripts/gitlab-rpc-docs.sh]: I would doubt that user exists. I just put that there as a placeholder for now. You will need to create a user in gitlab for the runner to use. Can then update those lines with the correct email and username.

File: .gitlab-ci.yml (hunk 1 above)
Comment (cpurta) [reply .gitlab-ci.yml]: I agree. Having that for every commit to master would get annoying pretty quick.

File: scripts/gitlab-rpc-docs.sh (hunk 2 above)
Comment (5chdn) [reply scripts/gitlab-rpc-docs.sh]: @General-Beck please suggest what we should use here :)

**Comment by 5chdn** [comment]
Scripts LGTM.

I created a test pipeline, let's see how it works out. https://gitlab.parity.io/parity/parity/pipelines/21869

**Comment by 5chdn** [comment]
Failure unrelated. Should be fixed now: https://gitlab.parity.io/parity/parity/pipelines/21887

**Comment by cpurta** [comment]
Looks like that job failed since the repo is called parity and not parity-ethereum when trying to copy. Will update the script.

**Comment by debris** [comment]
@5chdn please merge if it looks good to you :)

**Comment by cpurta** [comment]
@5chdn anything else needed for this PR?

**Comment by 5chdn** [comment]
Sorry, I was sick, I'll restart the CI once again.

https://gitlab.parity.io/parity/parity/pipelines/22544

**Comment by 5chdn** [comment]
docs failed: https://gitlab.parity.io/parity/parity/-/jobs/95442

```
fs.js:904
  return binding.readdir(pathModule._makeLong(path), options.encoding);
                 ^

Error: ENOENT: no such file or directory, scandir '/home/gitlab-runner/builds/4e7a1ea4/0/parity/parity/jsonrpc/.parity/rpc/src/v1/traits'
    at Object.fs.readdirSync (fs.js:904:18)
    at getMethodsFromRustTraits (/home/gitlab-runner/builds/4e7a1ea4/0/parity/parity/jsonrpc/scripts/helpers/parsed-rpc-traits.js:63:6)
    at Object.<anonymous> (/home/gitlab-runner/builds/4e7a1ea4/0/parity/parity/jsonrpc/scripts/helpers/parsed-rpc-traits.js:70:1)
    at Module._compile (module.js:652:30)
    at Object.Module._extensions..js (module.js:663:10)
    at Module.load (module.js:565:32)
    at tryModuleLoad (module.js:505:12)
    at Function.Module._load (module.js:497:3)
    at Module.require (module.js:596:17)
    at require (internal/module.js:11:18)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @parity/jsonrpc@2.1.6 build:markdown: `node scripts/build-rpc-markdown.js`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the @parity/jsonrpc@2.1.6 build:markdown script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/gitlab-runner/.npm/_logs/2018-08-21T19_04_30_599Z-debug.log
ls: cannot access 'jsonrpc/docs': No such file or directory
Switched to a new branch 'rpcdoc-update-v2.1.0-rc2'
On branch rpcdoc-update-v2.1.0-rc2
nothing to commit, working directory clean
On branch rpcdoc-update-v2.1.0-rc2
nothing to commit, working directory clean
error: Terminal is dumb, but EDITOR unset
Please supply the message using either -m or -F option.
fatal: could not read Username for 'https://github.com': No such device or address
ERROR: Job failed: exit status 1
```

**Comment by cpurta** [comment]
It appears to be due to not correctly copying the parity folder into the jsonrpc repo that is cloned. Working on a fix now.

**Comment by cpurta** [comment]
Fixed that error and a few others that popped up when running the job with a local gitlab CI runner. One thing that I forgot about was that this change will need a github api token environment variable needed in Gitlab CI settings (referenced as GITHUB_TOKEN in the script). This allows for the runner to be able to push code using the script. The script will fail until there is a valid token present on the runner.

**Comment by 5chdn** [comment]
Ok, I'll merge this now as-is and we will fix the final bit together with #8765

Thanks

**Pull request merged**